tests/components/deconz/test_binary_sensor.py

```diff
@@ -182,3 +182,24 @@ async def test_add_new_binary_sensor(hass):
 
     presence_sensor = hass.states.get("binary_sensor.presence_sensor")
     assert presence_sensor.state == "off"
+
+
+async def test_add_new_binary_sensor_ignored(hass):
+    """Test that adding a new binary sensor is not allowed."""
+    gateway = await setup_deconz_integration(
+        hass,
+        options={deconz.gateway.CONF_ALLOW_NEW_DEVICES: False},
+    )
+    assert len(hass.states.async_all()) == 0
+
+    state_added_event = {
+        "t": "event",
+        "e": "added",
+        "r": "sensors",
+        "id": "1",
+        "sensor": deepcopy(SENSORS["1"]),
+    }
+    gateway.api.event_handler(state_added_event)
+    await hass.async_block_till_done()
+
+    assert len(hass.states.async_all()) == 0
```

Comment:
Also check that this entity isn't in the entity registry.

Reply:
Missed that, yes will do!